queries/c_sharp/textobjects.scm

```diff
@@ -196,3 +196,64 @@
 
 ; comments
 (comment) @comment.outer
+
+; assignments
+(field_declaration
+  (_)
+  (variable_declaration
+    type: (_)
+    (variable_declarator
+      name: (_) @assignment.lhs
+      (_) @assignment.rhs) @assignment.inner)) @assignment.outer
+
+(field_declaration
+  (variable_declaration
+    type: (_)
+    (variable_declarator
+      name: (_) @assignment.lhs
+      (_) @assignment.rhs) @assignment.inner)) @assignment.outer
+
+(field_declaration
+  (_)
+  (variable_declaration
+    type: (_)
+    (variable_declarator
+      name: (_) @assignment.lhs @assignment.inner))) @assignment.outer
+
+(local_declaration_statement
+  (_)
+  (variable_declaration
+    type: (_)
+    (variable_declarator
+      name: (_) @assignment.lhs
+      (_) @assignment.rhs) @assignment.inner)) @assignment.outer
+
+(local_declaration_statement
+  (variable_declaration
+    type: (_)
+    (variable_declarator
+      name: (_) @assignment.lhs
+      (_) @assignment.rhs) @assignment.inner)) @assignment.outer
+
+(local_declaration_statement
+  (_)
+  (variable_declaration
+    type: (_)
+    (variable_declarator
+      name: (_) @assignment.lhs @assignment.inner))) @assignment.outer
+
+(expression_statement
+  (assignment_expression
+    left: (_) @assignment.lhs
+    right: (_) @assignment.rhs)) @assignment.inner @assignment.outer
```

Comment:
Inner and outer should not mean the same thing. Inner refers to "either" LHS or RHS, outer means select both. Sorry, this isn't documented very clearly as of now.

Reply:
Ok, I will fix these queries and add test cases.